blog/posts/announcing-melange-5.md

```diff
@@ -0,0 +1,285 @@
+we introduced support for 2 attributes in OCaml types that define variants:
+
+### `@mel.as`
+
+Specifying `[@mel.as ".."]` changes the variant emission in JavaScript to that
+string value.
+
+```js
+type t =
+  | [@mel.as "World"] Hello;
+
+let t = Hello
+```
+
+```js
+const t = /* Hello */ "World";
+```
+
+### `@mel.tag`
+
+A `@mel.as` variant type combined with `@mel.tag` allows expressing
+discriminated unions in an unobtrusive way:
+
+```js
+[@mel.tag "kind"]
+type t =
+  | [@mel.as "Foo"] Foo({ a: string, b: string, })
+  | [@mel.as "Bar"] Bar({ c: string, d: string, });
+
+let x = Foo({ a: "a", b: "b", });
+
+let y = Bar({ c: "c", d: "d", });
+```
+
+The Reason code above produces the following JavaScript:
+
+```js
+const x = {
+  kind: /* Foo */ "Foo",
+  a: "a",
+  b: "b"
+};
+
+const y = {
+  kind: /* Bar */ "Bar",
+  c: "c",
+  d: "d"
+};
+```
+
+In summary:
+
+- `[@mel.tag "kind"]` specifies that each variant containing a payload should
+  be tagged with `"kind"`.
+- the `[@mel.as ".."]` attribute in each variant type specifies what that
+  payload should be for each branch of the variant type.
+
+## `@mel.send` is way, way better
+
+When binding to methods of an object in JavaScript, Melange has historically
+supported 2 different ways of achieving the same: `@mel.send` and
+`@mel.send.pipe`. The only real reason why 2 constructs existed to do the same
+was to support two alternatives for chaining them in OCaml:
+[pipe-first](https://melange.re/v5.0.0/language-concepts.html#pipe-first) and
+[pipe-last](https://melange.re/v5.0.0/language-concepts.html#pipe-last). But
+this always felt like an afterthought, and code using `@mel.send.pipe` never
+felt intuitive to look at (e.g. in `external say: unit [@mel.send.pipe: t]`,
+one had to mentally place the `t` before `unit`, since the real signature is `t
+-> unit`).
+
+In Melange 5, we wanted to remove this weird split and further reduce the
+cognitive overhead of writing bindings to call JavaScript methods on an object
+or instance.
+
+We're introducing a way to mark the "self" instance argument with
+`@mel.this` and recommending only the use of `@mel.send` going forward.
+Starting from this release, `@mel.send.pipe` has been deprecated, and will be
+removed in the next major release of Melange. Here's an example:
+
+```js
+[@mel.send]
+external push: (~value: 'a=?, [@mel.this] array('a)) => unit = "push";
+
+let () = push([||], ~value=3);
+```
+
+The code above marks the `array('a)` argument as the instance to call the
+`push` method, which produces the following JavaScript:
+
+```js
+[].push(3);
+```
+
+Besides being more versatile, having an explicit marker with `@mel.this` is
+also more visually intuitive: when scanning Melange code containing `external`
+bindings, it becomes easier to spot which is the "this" argument. This feature
+is 100% backwards compatible with `@mel.send`: in the absence of `@mel.this`,
```

Comment:
The 100% read strange, either a feature like this is backwards compatible or it's not 😇